pkg/trace/trace.go

```diff
@@ -0,0 +1,102 @@
+package trace
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/valyala/fastrand"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var envs []string = []string{"prod", "dev"}
+var ar_dc []string = []string{"LON", "PAR", "DEN", "PHX", "DFW", "ATL", "BOS", "MHT", "JFK", "CAR", "LHR", "AMS", "FRA", "BOM", "CAL", "DEL", "MAD", "CHE", "FGH", "RTY", "UIO", "MHJ", "HAN", "YHT", "YUL", "MOL", "FOS", "KUN", "SRI", "FGR", "SUN", "PRI", "TAR", "SAR", "ADI", "ERT", "ITR", "DOW", "UQW", "QBF", "POK", "HQZ", "ZAS", "POK", "LIP", "UYQ", "OIK", "TRU", "POL", "NMC", "AZQ"}
+
+func StartTraceGeneration(file string, numTraces int, maxSpans int) {
+	log.Println("Starting generation of ", numTraces, " traces...")
+	traceCounter := 0
+	log.Println("Opening file ", file, "...")
+	f, err := os.Create(file)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+	for traceCounter < numTraces {
+		var traceId = uuid.NewString()
+		var randomNumberOfSpans = 1 + fastrand.Uint32n(uint32(maxSpans))
+		spanCounter := 0
+		for spanCounter < int(randomNumberOfSpans) {
+			span := generateChildSpan(traceId)
+			span["traceID"] = traceId
+			bytes, _ := json.Marshal(span)
+			_, err2 := f.Write(bytes)
+			if err2 != nil {
+				log.Fatal(err2)
+			}
+			spanCounter += 1
+		}
+		traceCounter += 1
+	}
+	f.Close()
+}
+
+func generateChildSpan(traceId string) map[string]interface{} {
+	span := make(map[string]interface{})
+	spanId := uuid.NewString()
+	span["spanID"] = spanId
+	span["startTime"] = time.Now()
+	span["startTimeMillis"] = time.Now().UnixNano() / int64(time.Millisecond)
+	span["flags"] = 1
+	span["operationName"] = " HTTP GET"
+	span["duration"] = fastrand.Uint32n(1_000)
+	span["tags"] = generateTagBody(10)
+	span["process"] = generateProcessTagsBody(10)
+	span["references"] = generateReferenceBody(traceId, spanId)
```

Comment:
we may have to add the key `"refType"` which is either `CHILD_OF` or `FollowsFrom`

Check out this diagram: https://docs.lightstep.com/docs/understand-distributed-tracing#context 

We would need to first generate a span that has no parent spans, and represents the trace a single request. 

Then, for the first random span, we create a `refType: CHILD_OF` for the above trace id. Then all other random spans can do a `refType: FollowsFrom` the previous span id. 

This way we are synthetically generating a trace that has many sequential spans.